### PR TITLE
fix:httpcode

### PR DESCRIPTION
**Pull request by constwz**
### Description

add a description of your changes here...

### Rationale

tell us why we need these changes...

### Example

add an example CLI or API response...

### Changes

Notable changes: 
* add each change in a bullet point here
* ...

### Potential Impacts
* add potential impacts for other components here
* ...

### CODE REVIEW

File: base/types/gfsperrors/error.go [hunk 1]
```diff
@@ -43,6 +46,23 @@ func init() {
 	})
 }
 
+func GetGfSpErr(err error) *GfSpError {
```
Comment (ruojunm) [inline base/types/gfsperrors/error.go:49]: what if the passed-in err is already a GfSpError?

Comment (constwz) [reply base/types/gfsperrors/error.go:49]: Never mind. The.Error() method will also return a parsable string, which will then be parsed to form a GfSpError

File: base/types/gfsperrors/error.go [hunk 2]
```diff
@@ -43,6 +46,26 @@ func init() {
...
+		return nil
+	}
+	if gfspErr, ok := err.(*GfSpError); ok {
+		return gfspErr
+	}
+	i := strings.Index(err.Error(), "desc = ")
+	if len(err.Error())-1 < i+6 {
+		return nil
+	}
+	errInfo := err.Error()[i+6:]
```
Comment (sysvm) [inline base/types/gfsperrors/error.go:60]: Does `[i+6:]` fit all cases?

Comment (constwz) [reply base/types/gfsperrors/error.go:60]: The result after the rpc call is "rpc error: code = Unknown desc = xxx" We only focus on xxx

**Pull request closed without merging**